nbgitpuller/static/js/index.js

```diff
@@ -42,6 +42,9 @@ GitSync.prototype.start = function() {
     if (typeof this.branch !== 'undefined' && this.branch != undefined) {
         syncUrlParams['branch'] = this.branch;
     }
+    if (typeof this.content_provider !== 'undefined' && this.content_provider != undefined) {
+        syncUrlParams['content_provider'] = this.content_provider;
```

Comment:
Tricky hmmm... In .js we use camelCase, in Python we use snake_case, and in the query line parameters we have had `baseUrl` in the past as a precedence, and that may be read both by js and python logic etc.

I think in the JS, we should stick to the JS typical camelCase though, regarding the query parameters, hmm... I'll ask in the Gitter chat jupyterhub/random for input about this. I think its most important to stay consistent, so perhaps I'm thinking we should stick with a consistency with baseUrl, at the same time, targetpath is all lowercase so we are already inconsistent no matter what we do.

```suggestion
    if (typeof this.contentProvider !== 'undefined' && this.contentProvider != undefined) {
        syncUrlParams['content_provider'] = this.contentProvider;
```

Reply:
yes I was losing myself in the context in terms of camelCase versus snake_case -- clearly missed a few as well.